### PR TITLE
feat(wallet): allow to avoid using unconfirmed UTXOs in transactions

### LINKED ISSUE #1832: [Wallet] Only allow transacting when superblock-confirmed balance is greater than 0
At the moment the Wallet checks the total balance when sending a vtt or a DR. However the wallet should check the available balance, meaning the balance that has been confirmed in a superblock, and ignore the pending balance.

### PR DESCRIPTION
**Pull request by tmpolaczyk**
Close #1832 

Keep track of the transactions that modify the utxo set but are not confirmed yet, so that we can ignore unconfirmed utxo when selecting utxos for a new transaction.

This feature is disabled by default, can be enabled by setting

    wallet.use_unconfirmed_utxos = false

in witnet.toml

### CODE REVIEW

File: wallet/src/repository/wallet/mod.rs
Comment (lrubiorod) [inline wallet/src/repository/wallet/mod.rs:45]: fmt :P

File: wallet/src/repository/wallet/mod.rs
Comment (lrubiorod) [inline wallet/src/repository/wallet/mod.rs:1216]: println?

Comment (tmpolaczyk) [reply wallet/src/repository/wallet/mod.rs:1216]: For debugging :smiley:

**Comment by tmpolaczyk** [comment]
This PR is blocked because the "available" balance does not match the balance that is calculated using the confirmed UTXOs. For example, the available balance can show 5000 nanowits and creating a transaction with less than that value can fail.

File: wallet/src/actors/app/error.rs
Comment (tmpolaczyk) [inline wallet/src/actors/app/error.rs:129]: This was the easiest way to send an object to the frontend, but this json object is serialized as a string, so the result is really ugly, JSON inside JSON:

    {"code":400,"message":"Validation Error","data":[["{\"available_balance\":17221973255,\"total_balance\":17221973255,\"transaction_value\":18000000853}","Wallet account has not enough balance"]]}

I don't know how to add additional fields to this error message, so any better alternatives welcome.

File: .github/workflows/push.yml
Comment (tmpolaczyk) [inline .github/workflows/push.yml:22]: Reminder to remove this commit before merging, this PR should go after #1888

**Comment by mariocao** [comment]
What's the status of this PR?

File: wallet/src/repository/wallet/mod.rs
Comment (mariocao) [inline wallet/src/repository/wallet/mod.rs:76]: We are still not sure if this if something a user would like or not... :smile: 

- This feature will make that transactions that require some unconfirmed UTXO, will have to wait at least 20 blocks.
- Some users might NOT want to able to create transactions from unconfirmed UTXOs (e.g. trades)
- Other users might prefer to have the freedom to create several transaction without having to wait for confirmation.

File: wallet/src/repository/wallet/mod.rs
Comment (lrubiorod) [inline wallet/src/repository/wallet/mod.rs:1238]: Is it not the opposite logic? It means, if we want to use unconfirmed_utxos we include the pending transactions

Comment (lrubiorod) [reply wallet/src/repository/wallet/mod.rs:1238]: I didnt say nothing, it is to create the hashset to check if we can use or not, so it is ok

Comment (tmpolaczyk) [reply wallet/src/repository/wallet/mod.rs:1238]: It seems backwards but this is the logic: `unconfirmed_transactions` is the list of transactions that will not be used. If `unconfirmed_transactions` is empty, we will use all the unconfirmed UTXOs (this is the old behavior before this PR). If `unconfirmed_transactions` is equal to pending transactions, we will not use unconfirmed UTXOs in new transactions.

**Review by lrubiorod — APPROVED** [review]

**Pull request merged**
